go/libraries/doltcore/sqle/dtablefunctions/dolt_log.go

```diff
@@ -14,11 +14,16 @@
 
 package dtablefunctions
 
+// This table function violates SQL analyzer principles by evaluating expressions
+// during both analysis and execution phases. This is necessary because CLI
+// arguments requires pre-evaluated strings for schema determination. Bind variables
+// are handled by skipping evaluation during analysis and deferring to execution.
```

Comment:
I think you can move this down to the `evalArguments` function, or somewhere else nearby where the expressions are eval'ed. Also, so some minor changes suggested: 
```suggestion
// This table function violates SQL analyzer principles by evaluating expressions
// during analysis. This is necessary because the schema changes based on what
// arguments are supplied (e.g. --parent), and the schema needs to be known 
// during analysis time. Bind variables are skipped over during the initial analysis of 
// the prepared statement, and get fully resolved when they are bound when the 
// prepared statement is later executed.
```